src/components/ReceiptModal/ReceiptModal.jsx

```diff
@@ -0,0 +1,223 @@
+import * as React from "react"
+import { useState } from "react"
+import { Link } from "react-router-dom"
+import Modal from "react-bootstrap/Modal"
+import { BsDot } from 'react-icons/bs'
+import { AiOutlinePlus, AiOutlineMinus, AiOutlineClose } from 'react-icons/ai'
+import { Container, Row, Col, Button, Alert, Form } from "react-bootstrap"
+import PacmanLoader from "react-spinners/PacmanLoader";
+import { useEffect } from "react"
+import axios from 'axios';
+import { ref, uploadBytesResumable, getDownloadURL, uploadBytes } from "firebase/storage"
+import { storage } from "../../firebase"
+import { units, basicUnits, convertToStandard, updateStandardAmount, isVolumeUnit, basicCategories } from "../../utils/conversion"
+import "./ReceiptModal.css"
+
+export default function ReceiptModal(props) {
+    const receiptScannerUrl = "http://localhost:3001/receipt?"
+    const [error, setError] = useState("")
+    const [curImageFile, setCurImageFile] = useState(null)
+    const [curImageUrl, setCurImageUrl] = useState("")
+    const [loadingPercent, setLoadingPercent] = useState(0)
+    const [receiptItems, setReceiptItems] = useState([])
+    const [uploadingImage, setUploadingImage] = useState(false)
+    const [scanningImage, setScanningImage] = useState(false)
+    const { userProducts, setUserProducts, ...modalProps } = props
+
+    useEffect(() => {
+        if (curImageFile != null) {
+            setUploadingImage(true)
+            const imageRef = ref(storage, `images/${curImageFile.name}`)
+            const uploadTask = uploadBytesResumable(imageRef, curImageFile);
+            uploadTask.on(
+                "state_changed",
+                (snapshot) => {
+                    const percent = Math.round(
+                        (snapshot.bytesTransferred / snapshot.totalBytes) * 100
+                    );
+
+                    // update progress
+                    setLoadingPercent(percent);
+                },
+                (err) => console.log(err),
+                () => {
+                    // download url
+                    getDownloadURL(uploadTask.snapshot.ref).then((url) => {
+                        console.log(url);
+                        setCurImageUrl(url)
+                    });
+                    setUploadingImage(false)
+                }
+            );
+        }
+    }, [curImageFile])
+
+    const onImageChange = (event) => {
+        let img = event.target.files[0];
+        setCurImageFile(img)
```

Comment:
Nit: Can you not make this: setCurImageFile(event.target.files[0]) and remove img?